src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -13,8 +13,12 @@
     /// </summary>
     internal class HostingDiagnosticListener : IApplicationInsightDiagnosticListener
     {
+        /// <summary>
+        /// Conversion from stopwatch time (ticks since boot) to local time.
+        /// </summary>
+        private static readonly DateTimeOffset stopwatchEpoch = DateTime.Now - TimeSpan.FromTicks(Stopwatch.GetTimestamp());
```

Comment:
Please use Start/Stop.

I was planning to go in production with v2.0.0 and realized the bug with the RequestTelemetry and DependencyTelemetries not correlating the OperationId correctly and an issue with the RequestTelemetries timestamp being set after the one set on the related DependencyTelemetry instances.

I was about to implement my own fix but noticed your pull request which brings in the OperationCorrelationTelemetryInitializer. I also noticed the changes you made to the HostingDiagnosticListener.
I cloned your pull request branch in hope of it fixing the timing issues, but it is worse (at least when testing locally) since Stopwatch.GetTimestamp() returns the timestamp of my PC’s last boot.

Using the RequestTelemetry Start/Stop extension instead of the timespan rectified the situation on my end.
I understand wanting to use the OnBeginRequest and OnEndRequest timestamp parameters, but this just not seem to be working on my end. I am without doubt missing key elements here to fully grasp the full impact of this choice, but I fail to see how using Stopwatch.GetTimestamp() will give consistent, testable results.

Since IApplicationInsightDiagnosticListener is internal, I cannot simply remove from the services collection your implementation and add in my own using Start/Stop.